.github/workflows/docker.yml

```diff
@@ -0,0 +1,71 @@
+# GitHub actions workflow which builds and publishes the docker images.
+
+name: Build docker images
+
+on:
+  push:
+    branches: [ master ]
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  packages: write
+jobs:
+  build:
+    if: github.repository == 'DarkEnergyProcessor/NPPS4'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up QEMU
+        id: qemu
+        uses: docker/setup-qemu-action@v3
+        with:
+          platforms: arm64
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Inspect builder
+        run: docker buildx inspect
+
+      - name: Install Cosign
+        uses: sigstore/cosign-installer@v3.5.0
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Extract version from Cargo.toml
+        shell: bash
+        run: |
+          echo "NPPS4_VERSION=0.1" >> $GITHUB_ENV
+
+      - name: Log in to DockerHub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Calculate docker image tag
+        id: set-tag
+        uses: docker/metadata-action@master
+        with:
+          images: |
+            docker.io/MikuAuahDark/NPPS4
```

Comment:
I guess I can replace `MikuAuahDark` with `${{ secrets.DOCKERHUB_USERNAME }}`?

Reply:
for other people to pull from docker, you kinda have to give them the user name anyway

Reply:
Right, but let's say I **don't** want to create a Docker account in the first place.

Reply:
Just create a throwaway account I suppose?
Maybe if you don't want to interact with Docker at all, the docker image can be published to github instead?

Reply:
In that case I'll look on this again next week.